core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -787,19 +770,22 @@ class ReplicaManager(val config: KafkaConfig,
             )
         }
 
-        val errorResults = errorsPerPartition.map {
-          case (topicPartition, error) =>
-            topicPartition -> LogAppendResult(
-              LogAppendInfo.UNKNOWN_LOG_APPEND_INFO,
-              Some(error.exception())
+        val allResults = localProduceResults ++ errorResults
+        val produceStatus = allResults.map { case (topicPartition, result) =>
+          topicPartition -> ProducePartitionStatus(
+            result.info.lastOffset + 1, // required offset
+            new PartitionResponse(
+              result.error,
+              result.info.firstOffset,
+              result.info.lastOffset,
+              result.info.logAppendTime,
+              result.info.logStartOffset,
+              result.info.recordErrors,
+              result.exception.map(_.getMessage).orNull
```

Comment:
Are we going to start returning the error message for other exceptions now too? I think we will add the default exception message when previously we were not.

Reply:
Hmm, good catch. somehow I didn't think about that. I refactored some of the LogAppendResult stuff to try to make it more explicit when we use a custom error message.